pkg/datarepair/checker/checker.go

```diff
@@ -92,13 +111,20 @@ func (c *checker) identifyInjuredSegments(ctx context.Context) (err error) {
 					return Error.New("error getting offline nodes %s", err)
 				}
 				numHealthy := len(nodeIDs) - len(missingPieces)
-				if int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold {
+				if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) < pointer.Remote.Redundancy.RepairThreshold) {
 					err = c.repairQueue.Enqueue(&pb.InjuredSegment{
 						Path:       string(item.Key),
 						LostPieces: missingPieces,
 					})
 					if err != nil {
 						return Error.New("error adding injured segment to queue %s", err)
+					} else if int32(numHealthy) < pointer.Remote.Redundancy.MinReq {
```

Comment:
I believe this `else if` statement on line 121 needs to be moved below the initial `if` statement. Like this:
```
if (int32(numHealthy) >= pointer.Remote.Redundancy.MinReq) && (int32(numHealthy) < 
pointer.Remote.Redundancy.RepairThreshold) {
     err = c.repairQueue.Enqueue(&pb.InjuredSegment{
          Path:       string(item.Key),
          LostPieces: missingPieces,
     })
     if err != nil {
          return Error.New("error adding injured segment to queue %s", err)
     }
} else if int32(numHealthy) < pointer.Remote.Redundancy.MinReq {
```

Reply:
Done